### PR TITLE
[CP-1079] Implement architecture of Outbox endpoint

### PR DESCRIPTION
**Pull request by dkarski**
Jira: [CP-1079]

**Description**

<details>
<summary><b>Screenshots</b></summary>
// put images here
</details>

[CP-1079]: https://appnroll.atlassian.net/browse/CP-1079?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/data-sync/repositories/contact.repository.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+// TODO: fully implementation will be https://appnroll.atlassian.net/browse/CP-1049
+export class ContactRepository {
+  constructor() {}
```
Comment (igorbezsmertnyi) [inline packages/app/src/data-sync/repositories/contact.repository.ts:8]: The repository should have a model as a dependency

File: packages/app/src/outbox/services/outbox.service.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+    for (const entry of entries) {
+      const handle = this.entryHandlersMap[entry.type] ?? asyncNoop
+      await handle(entry)
+    }
+
+    await this.deleteOutboxEntriesRequest(entries.map(({ uid }) => uid))
+
+    await this.ipc.sendToRenderers(IpcEvent.DataLoaded)
```
Comment (igorbezsmertnyi) [inline packages/app/src/outbox/services/outbox.service.ts:61]: I think this event can be emitted from the observer

Comment (igorbezsmertnyi) [reply packages/app/src/outbox/services/outbox.service.ts:61]: It's reduce amount of dependencies in this service

File: packages/pure/src/device/device.types.ts [hunk 3]
```diff
@@ -71,6 +71,7 @@ export enum Endpoint {
   Messages = 8,
   CallLog = 9,
   Security = 13,
+  Outbox = 13,
```
Comment (igorbezsmertnyi) [inline packages/pure/src/device/device.types.ts:74]: Is the outbox endpoint have the same name as security endpoint?

Comment (dkarski) [reply packages/pure/src/device/device.types.ts]: [c4dd030](https://github.com/mudita/mudita-center/pull/670/commits/c4dd03028534a46de10837e02f5a176e65a5903d)

File: packages/app/src/outbox/services/outbox.service.ts (hunk 2 above)
Comment (dkarski) [reply packages/app/src/outbox/services/outbox.service.ts]: [8eb4cd0](https://github.com/mudita/mudita-center/pull/670/commits/8eb4cd036c1d4b0110ede1de9e75605d73a4f95f)

File: packages/app/src/data-sync/repositories/contact.repository.ts (hunk 1 above)
Comment (dkarski) [reply packages/app/src/data-sync/repositories/contact.repository.ts:8]: will be done in https://appnroll.atlassian.net/browse/CP-1049

**Comment by igorbezsmertnyi** [comment]
@dkarski Looks good, but fix the test, please 🙂

**Comment by dkarski** [comment]
> @dkarski Looks good, but fix the test, please 🙂

@igorbezsmertnyi done 🙃 😉

**Review by igorbezsmertnyi — APPROVED** [review]

**Pull request merged**